recipes/grpc/all/test_package/conanfile.py

```diff
@@ -31,6 +31,8 @@ def build(self):
             return
         with self._buildenv():
             cmake = CMake(self)
+            if not tools.valid_min_cppstd(self, 11):
+                cmake.definitions["CMAKE_CXX_STANDARD"] = 11
```

Comment:
Use same approach used to build the library itself

Reply:
that looks weird... also, it's no longer recommended to use as I understand

Reply:
The best approach would be to patch upstream CMakeLists to use compile feature, but it's not straightforward.

Reply:
but it's test package. we can use `target_compile_features` in out own CMakeLists.txt, right?

Reply:
yes

Reply:
We use this approach to build the library, because otherwise it would force C++11 regardless of `compiler.cppstd` due to https://github.com/grpc/grpc/blob/v1.45.2/CMakeLists.txt#L226-L232. It's a tradeoff, better solution than unconditional C++11, but not the best one.

As I said, we should patch upstream CMakeLists to use `cxx_std_11` compile feature, and submit the patch upstream. `cxx_std_11` is tedious to inject with a patch when there are many targets (there is no global compile feature like CMAKE_CXX_STANDARD for PROPERTIES), and there are many gRPC versions to patch in conan-center.

Reply:
I would say, how exactly upstream manages C++ should be (ideally) a black box for us.
strictly speaking, the fact that library has to be built via `-std=c++11` doesn't necessariily mean test_package should consume it via `-std=c++11`.
for instance, in case of `abseil`, we build it as `-std=c++11`, but `test_package` verifies it can be perfectly consumed with 11/14/17/20/23: https://github.com/conan-io/conan-center-index/blob/master/recipes/abseil/all/test_package/CMakeLists.txt#L13-L35
in general case, even if library is built via some C++ standard, doesn't mean it has it standard as a API/ABI requirement, it might be completely orthogonal.

Reply:
btw, good news. `compiler.cppstd` will be a mandatory in conan v2, or in other words, it won't be set to `None`, but will be explicitly set to some standard (preferrably, the default of the compiler, something reported by `__cplusplus` without specifying any `-std`).